Fix removing extra space from event

AnthonyGuerreiro opened the pull request:
If server sends the following event:
event:FooBar
data:{"foo":"bar baz"}

The first space after the first ':' is removed, becoming:
{"foo":"barbaz"}

This commit should first check if the second part of the line (after ':') starts with a white space.
If it doesn't then it won't replace any space in its content.

I assume this exists to remove the space for cases when the input is in the format:
data: {"foo":"bar baz"}

jkodumal commented:
Thanks @AnthonyGuerreiro -- good catch. Would you mind adding a few test cases?

jkodumal commented on src/main/java/com/launchdarkly/eventsource/EventParser.java:
```diff
@@ -43,7 +43,10 @@ public void line(String line) {
       String field = line.substring(0, colonIndex);
-      String value = line.substring(colonIndex + 1).replaceFirst(" ", EMPTY_STRING);
+      String value = line.substring(colonIndex + 1);
+      if (value.charAt(0)==' ') {
```
Note for posterity's sake: this is the correct behavior according to the spec:

https://www.w3.org/TR/eventsource/#parsing-an-event-stream

AnthonyGuerreiro commented:
Actually, I introduced a bug with this fix. I fixed it already in my branch but you were quicker than me.
Not sure if I can somehow add an extra commit to this PR or need to do another one?

jkodumal commented:
Feel free to open another PR.

AnthonyGuerreiro commented:
Done (#26 ). Thank you!
